domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -37,4 +39,14 @@ public async Task<ActionResult> Create([FromBody] CreateTransferAgreement reques
 
         return Created($"api/transfer-agreements/{result.Id}", result);
     }
+
+    [HttpGet("{id}")]
+    public async Task<ActionResult> Get([FromRoute] Guid id)
+    {
+        var tin = User.FindFirstValue("tin")!;
+        var subject = User.FindSubjectClaim();
+
+        var result = await transferAgreementRepository.GetTransferAgreement(id, subject, tin);
+        return result == null ? NotFound() : Ok(result);
```

Comment:
You are returning the domain entity/business object/database model. Do you not want to map it to a dedicated API response model?

Reply:
Also note that the API should communicate all timestamps/dates in Unix Time Seconds